src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -227,16 +228,37 @@ await InvokeAsync(() =>
 
         void SetSelectedResourceOption(ResourceViewModel resource)
         {
-            Debug.Assert(_resources is not null);
-
-            PageViewModel.SelectedOption = _resources.Single(option => option.Id?.Type is not OtlpApplicationType.ResourceGrouping && string.Equals(ResourceName, option.Id?.InstanceId, StringComparison.Ordinal));
+            PageViewModel.SelectedOption = GetSelectedOption();
             PageViewModel.SelectedResource = resource;
 
             Logger.LogDebug("Selected console resource from name {ResourceName}.", ResourceName);
             loadingTcs.TrySetResult();
         }
     }
 
+    private SelectViewModel<ResourceTypeDetails> GetSelectedOption()
+    {
+        Debug.Assert(_resources is not null);
+        return _resources.SingleOrDefault(
```

Comment:
Can `ApplicationsSelectHelpers.GetApplication` be used here? That is what other pages with a selector (structured logs, traces, metrics) use to match the name to a select view model.

Reply:
Thanks, I don't know how I didn't find this.